ndc-models/src/schema.rs

```diff
@@ -212,6 +216,40 @@ pub enum AggregateFunctionDefinition {
 }
 // ANCHOR_END: AggregateFunctionDefinition
 
+// ANCHOR: ExtractionFunctionDefinition
+/// The definition of an aggregation function on a scalar type
+#[skip_serializing_none]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[schemars(title = "Extraction Function Definition")]
+pub struct ExtractionFunctionDefinition {
+    /// The result type, which must be a defined scalar types in the schema response.
+    pub result_type: ScalarTypeName,
+    /// The meaning of this extraction function
+    pub r#type: ExtractionFunctionType,
```

Comment:
This way of defining things does not match the spec's description of how this is supposed to work. In the spec, you said that all the (current) well-known `ExtractionFunctionType`s must return a scalar type of `intX` representation, but Custom functions can return amy type. This structure only allows the returning of Scalar Types.

To match what's defined in the spec text (and I think what the spec says is quite reasonable!), `result_type` needs to move into ExtractionFunctionType, for each variant that requires it, and Custom's `result_type` would be `Type`.

Either that or we relax `result_type` to be `Type` and enforce compliance in ndc-test.

Reply:
I've moved everything into one top level enum type, just like for aggregates.